File: app/src/main/java/com/ohmcoe/vultr/SetAPIKeyFragment.java
package com.ohmcoe.vultr;


import android.app.Fragment;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;


public class SetAPIKeyFragment extends Fragment {
    View setAPIKeyView;
    EditText txtAPIKey;
    Button btnOK;
    private String APIKey;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        setAPIKeyView = inflater.inflate(R.layout.fragment_set_apikey, container, false);

        txtAPIKey = (EditText)setAPIKeyView.findViewById(R.id.txtAPIKey);
        btnOK = (Button)setAPIKeyView.findViewById(R.id.btnOK);

        btnOK.setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View v) {
                updateAPIKey();
            }
        });

        Bundle bundle = getArguments();
        APIKey = bundle.getString("API-Key");
        txtAPIKey.setText(APIKey);

        return setAPIKeyView;
    }


    public void updateAPIKey()
    {
        String fileName = NavigationActivity.configFile;
        FileOutputStream outputStream;

        try {
            outputStream = getContext().openFileOutput(fileName, Context.MODE_PRIVATE);
            outputStream.write(txtAPIKey.getText().toString().getBytes());
            outputStream.close();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }

        Intent in = new Intent(getContext(), NavigationActivity.class);
        startActivity(in);
    }
}

File: app/src/main/java/com/ohmcoe/vultr/SetAPIKeyFragment.kt
package com.ohmcoe.vultr

import android.app.Fragment
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_set_apikey.view.*


class SetAPIKeyFragment : Fragment() {
    private var setAPIKeyView:View? = null

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val setAPIKeyView = inflater!!.inflate(R.layout.fragment_set_apikey, container, false)
        val settings = activity.getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE)
        val api:String? = settings.getString(MainActivity.API_KEY, null)


        setAPIKeyView.btnOK.setOnClickListener { updateAPIKey() }
        setAPIKeyView.txtAPIKey.setText(api)

        this.setAPIKeyView = setAPIKeyView
        return setAPIKeyView
    }

    fun updateAPIKey(): Unit {
        val sharePreferences = activity.getSharedPreferences(MainActivity.PREFS_NAME, Context.MODE_PRIVATE)
        val editor:SharedPreferences.Editor = sharePreferences.edit()

        editor.putString(MainActivity.API_KEY, setAPIKeyView!!.txtAPIKey.text.toString())
        editor.apply()
        editor.commit()


        val intent = Intent(activity, MainActivity::class.java)
        activity.finish()
        startActivity(intent)
    }
}
